src/ddo/DDO.ts

```diff
@@ -251,10 +337,21 @@ export class DDO {
     return metadataService.attributes.main
   }
 
-  public async updateService(nevermined: Nevermined, service: any): Promise<void> {
-    this.service[0] = service
+  /**
+   * Updates a service in the DDO
+   * @param service the service to be updated
+   * @param serviceIndex the position of the service in the DDO.services array
+   */
+  public updateService(service: any, serviceIndex = 0) {
+    this.service[serviceIndex] = service
```

Comment:
I'm concerned with having to rely on the services array to be sorted.

I think here we should update the service that is at index `findServiceById(serviceIndex)`  and throw an exception if the service does not exist?

And I think this behavior should be extended to `addService` and `replaceService` and we should stop using serviceIndex has the index of the array and instead always use `findServiceById`

Reply:
This methods was used in the code to update the `metadata`. I created a new `updateMetadataService` and used everywhere. This new method now checks the `metadata` service exists. 
I'm deprecating the `updateService` method but keeping it just in case we want to force a service replacement.